src/Microsoft.Kiota.Cli.Commons/Extensions/CommandBuilderExtensions.cs

```diff
@@ -69,4 +70,77 @@ public static CommandLineBuilder RegisterCommonServices(this CommandLineBuilder
         });
         return builder;
     }
+
+    /// <summary>
+    /// Registers a headers option to all executable commands in the builder
+    /// and a middleware that reads options from the parsed result and adds
+    /// them to the store instance returned after calling
+    /// <paramref name="headersStoreGetter"/>.
+    /// </summary>
+    /// 
+    /// <param name="builder">A command line builder</param>
+    /// <param name="headersStoreGetter">
+    /// A delegate that will be called to get an instance of
+    /// <see cref="IHeadersStore"/>.
+    /// </param>
+    /// <param name="name">
+    /// A custom name for the registered option.
+    /// Defaults to <c>--headers</c>
+    /// </param>
+    /// <param name="customDescription">
+    /// An optional description for the registered headers option.
+    /// </param>
+    /// 
+    /// <returns>
+    /// The same instance of <see cref="CommandLineBuilder"/>.
+    /// </returns>
+    /// 
+    /// <remarks>
+    /// <para>
+    /// This function must be called after the root command has been set in
+    /// the <see cref="CommandLineBuilder"/>.
+    /// </para>
+    /// <para>
+    /// This function does nothing to add the headers to a request. You would
+    /// need to implement a way to read from the updated
```

Comment:
would be great to have some convenience function here that binds this and the delegatinghandler. I think it's a little confusing for a person to know they need to add a handler for the http pipeline as well.

Reply:
I agree. I'll try to figure out a good way to do this.

Reply:
The issue I have right now is that the HttpClients are created outside of this library, but the handlers (on `System.Net.Http`) have to be added before instance creation.

Reply:
This will be implemented as part of the larger refactor work in microsoftgraph/msgraph-cli#323